src/tcache.c

```diff
@@ -189,7 +187,17 @@ tcache_bin_flush_large(tsd_t *tsd, tcache_bin_t *tbin, szind_t binind,
 		if (config_prof) {
 			idump = false;
 		}
+
 		malloc_mutex_lock(tsd_tsdn(tsd), &locked_arena->lock);
+		for (unsigned i = 0; i < nflush; i++) {
+			void *ptr = *(tbin->avail - 1 - i);
+			assert(ptr != NULL);
+			extent = iealloc(tsd_tsdn(tsd), ptr);
+			if (extent_arena_get(extent) == locked_arena) {
```

Comment:
Oh!  That's an interesting optimization idea.  I will create an issue to keep track of it.